.github/workflows/build-dev.yml

```diff
@@ -42,6 +42,34 @@ jobs:
           name: dist
           path: dist
 
+  build-storybook:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: "18"
+
+      - uses: actions/cache@v4
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node_modules-
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Build Storybook
+        run: npm run build-storybook
+
+      - name: Archive `storybook-static`
+        uses: actions/upload-artifact@v4
+        with:
+          name: storybook
+          path: storybook-static
+
```

Comment:
I'd recommend factoring this (and the `build` job) into their own files and invoking them from each of the deploy actions via
```
jobs:
  build:
    uses: ./.github/workflows/build.yml
```

It looks to me like the artifact upload/download will just continue to work in that case, though I haven't tested this myself.